test/dynamo/test_repros.py

```diff
@@ -1160,11 +1124,11 @@ def test_reformer_train(self):
             cnt = self._reformer(nopython=False)
         # cant inline torch.autograd.Function means graph break
         if torch._dynamo.config.assume_static_by_default:
-            self.assertExpectedInline(cnt.frame_count, """3""")
-            self.assertExpectedInline(cnt.op_count, """10""")
+            self.assertExpectedInline(cnt.frame_count, """1""")
+            self.assertExpectedInline(cnt.op_count, """5""")
         else:
-            self.assertExpectedInline(cnt.frame_count, """3""")
-            self.assertExpectedInline(cnt.op_count, """10""")
+            self.assertExpectedInline(cnt.frame_count, """1""")
+            self.assertExpectedInline(cnt.op_count, """5""")
```

Comment:
The reason of changing this unit test: The original backward implementation is wrong. It was not tested because we don't trace the backward function before this change. This was copied from modeling_reformer.py in huggingface, I found it's not easy to copy everything updated from there, so I made a simplification.